pyfahviewer/static/main.js

```diff
@@ -1,138 +1,124 @@
 (function() {
-  var leaderboard_template = null;
-  var slot_template = null;
-  var fetching = false;
+  const host = `${location.protocol}//${location.hostname}` + (location.port ? ':' + location.port : '');
 
+  var leaderboardTemplate = null;
+  var slotTemplate = null;
+  var fetching = false;
+  
   document.addEventListener("DOMContentLoaded", function () {
-    leaderboard_template = document.getElementById("leader-template").content.firstElementChild;
-    slot_template = document.getElementById("slot-template").content.firstElementChild;
-    load_data();
+    leaderboardTemplate = document.getElementById("leader-template").content.firstElementChild;
+    slotTemplate = document.getElementById("slot-template").content.firstElementChild;
+    loadData();
 
-    window.setInterval(load_data, 10000);
+    window.setInterval(loadData, 10000);
   });
 
-  function load_data() {
+  async function loadData() {
     if (fetching) {
-      console.log("Did not fetch because a fetch is already in progress.")
+      console.log("Did not fetch because a fetch is already in progress.");
       return false;
     }
 
     fetching = true;
-    done_fetching_team = false;
-    done_fetching_slots = false;
-
-    host = location.protocol+'//'+location.hostname+(location.port ? ':'+location.port: '');
-
-    // Get team stats
-    fetch(host + "/api/team")
-      .then((result) => {
-        handle_team_response(result);
-
-        done_fetching_team=true;
-        if (done_fetching_slots) {
-          fetching = false;
-        }
-      })
-
-    // Get slot info
-    fetch(host + "/api/slots")
-      .then((result) => {
-        handle_slot_response(result);
-
-        done_fetching_slots=true;
-        if (done_fetching_team) {
-          fetching = false;
-        }
-      })
+
+    await Promise.all([
+      refreshTeam(),
+      refreshSlots()
+    ]);
+
+    fetching = false;
   }
 
-  function handle_team_response(response) {
+  async function refreshTeam(response) {
+    let response = await fetch(`${host}/api/team`);
```

Comment:
`response`'s scope is already set since it's an argument. This redeclaration breaks it.

Reply:
frick